controllers/replication-controller/dellcsireplicationgroup_controller.go

```diff
@@ -478,3 +511,319 @@ func (r *ReplicationGroupReconciler) SetupWithManager(mgr ctrl.Manager, limiter
+	}
+
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(pvcs.Items))
+
+	for _, pvc := range pvcs.Items {
+		wg.Add(1)
+		go func(pvc v1.PersistentVolumeClaim) {
+			defer wg.Done()
+			err := r.swapPVC(ctx, c, pvc.Name, pvc.Namespace, pvc.Annotations[controller.RemotePV], rgTarget, log)
+			if err != nil {
+				errChan <- fmt.Errorf("error swapping PVC %s/%s: %s", pvc.Namespace, pvc.Name, err)
+			}
+		}(pvc)
+	}
+
+	go func() {
+		wg.Wait()
+		close(errChan)
+	}()
+
+	var errs []error
+	for err := range errChan {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("errors occurred while swapping PVCs: %s", errs)
+	}
+
+	return nil
+}
+
+func (r *ReplicationGroupReconciler) swapPVC(ctx context.Context, client connection.RemoteClusterClient, pvcName, namespace, targetPV, rgTarget string, log logr.Logger) error {
+	// Read the PVC
+	pvc, err := client.GetPersistentVolumeClaim(ctx, namespace, pvcName)
+	if err != nil {
+		return fmt.Errorf("error getting pv %s: %s", pvcName, err)
+	}
+
+	// Save the Reclaim Policy for both PVs - return reclaim policy to makepvcreclaimpolicyretain
+	pv, err := client.GetPersistentVolume(ctx, pvc.Spec.VolumeName)
+	if err != nil {
+		log.V(common.InfoLevel).Info(fmt.Sprintf("Error retrieving local PV %s: %s", pvc.Spec.VolumeName, err.Error()))
+		return err
+	}
+	localPVPolicy := pv.Spec.PersistentVolumeReclaimPolicy
+
+	pv, err = client.GetPersistentVolume(ctx, pvc.Annotations[controller.RemotePV])
+	if err != nil {
+		log.V(common.InfoLevel).Info(fmt.Sprintf("Error retrieving remote PV %s: %s", pvc.Annotations[controller.RemotePV], err.Error()))
+		return err
+	}
+	remotePVPolicy := pv.Spec.PersistentVolumeReclaimPolicy
+
+	// Make the local PV reclaim policy Retain
+	if err = makePVReclaimPolicyRetain(ctx, client, pvc.Spec.VolumeName, log); err != nil {
+		return err
+	}
+
+	// Make the remote PV reclaim policy Retain
+	if err = makePVReclaimPolicyRetain(ctx, client, pvc.Annotations[controller.RemotePV], log); err != nil {
+		return err
+	}
+
+	// Check that the request targetPV volume is our replica (remotePV)
+	remotePV := pvc.Annotations[controller.RemotePV]
+	if targetPV != remotePV {
+		err := fmt.Errorf("requested target %s doesn't match available replica %s", targetPV, remotePV)
+		log.V(common.InfoLevel).Info(err.Error())
+		return err
+	}
+
+	// Delete the existing PVC
+	err = client.DeletePersistentVolumeClaim(ctx, pvc)
+	if err != nil {
+		log.V(common.InfoLevel).Info(fmt.Sprintf("error deleting PVC %s: %s", pvcName, err.Error()))
+		return err
+	}
+
+	// Wait until PVC is deleted
+	done := false
+	for iteration := 0; !done; iteration++ {
+		time.Sleep(2 * time.Second)
+		_, err := client.GetPersistentVolumeClaim(ctx, namespace, pvcName)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				done = true
+			} else {
+				log.V(common.InfoLevel).Info(fmt.Sprintf("Error checking PVC %s/%s: %s", namespace, pvcName, err.Error()))
```

Comment:
Please update the log message to indicate we were waiting for the PVC to be deleted when the error occurred.